Mathlib/LinearAlgebra/FiniteSpan.lean

```diff
@@ -0,0 +1,37 @@
+/-
+Copyright (c) 2023 Oliver Nash. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Oliver Nash, Deepro Choudhury
+-/
+import Mathlib.GroupTheory.OrderOfElement
+import Mathlib.LinearAlgebra.Span
+
+/-!
+
+# Additional results about finite spanning sets in linear algebra
+
+-/
+
+open Set Function
+open Submodule (span)
+
+/-- A linear equivalence which preserves a finite spanning set must have finite order. -/
+lemma LinearEquiv.isOfFinOrder_of_finite_of_span_eq_top_of_image_subset
```

Comment:
I don't particularly mind the new file. But I could also imagine it finding a home in the other new file.